DOCPR-373: rule no. 003 correct use of Ubuntu versions

AnneCYH opened the pull request:
PR for DOCPR-373 to address rule no. 003, correct use of Ubuntu versions.
I don't have examples of what are common incorrect uses; hence am just making assumptions about the incorrect scenarios.

The list of releases are checked until 18.04 since the rest are historically very old versions already. This is due to the current regex definition which is not too flexible in terms of versions that are being detected (to be more accurate).

Also there is only 1 version of the test file (in MD) since for this rule, the syntax for RST vs MD is irrelevant.

Thanks.

keirthana reviewed (changes requested):
One minor nitpick: Can we also add "Ubuntu 24.04 LTS (Noble Numbat)" format to the examples in the message?

Otherwise, LGTM!

AnneCYH commented on en/index.md:
```diff
@@ -83,6 +83,7 @@ Some examples of correct usage:
 - **Ubuntu 22.10**
 - **Ubuntu 23.04 (Lunar Lobster)** _(note case!)_
 - **Ubuntu 23.10 (Mantic)**
+- **Ubuntu 24.04 LTS (Noble Numbat)**
```
Add the example for full version number + code name.

keirthana reviewed (dismissed):
LGTM!

AnneCYH commented:
@keirthana , thanks for the review!
I've added the example in the output 'suggestion' message and added the same naming style to the style guide itself.

AnneCYH commented on styles/Canonical/003-Ubuntu-names-versions.yml:
```diff
@@ -0,0 +1,29 @@
+  - 'Ubuntu (24|22|20|18)\.04(?! LTS)'
+  # Non-LTS releases should not be followed by "LTS" suffix.
+  - 'Ubuntu (23\.10|23\.04|22\.10|21\.10|21\.04|20\.10|19\.10|19\.04|18\.10|17\.10|17\.04) LTS'
+  # All releases should match the code name.
```
This regex checks for incorrect usages of Ubuntu. If the Ubuntu version is matched and a parentheses follows, the check will fail if anything other than the code name is not found in the parentheses.

evilnick commented:
This is really great - it goes beyond what i thought we would be able to do.
I have tweaked a few things
 - scope: I think we overuse the 'raw' scope which tends to throw up false positives. I tweaked it here and it still matches the tests
 - test: I expanded the test to include a heading and a table to check the scope was working

evilnick commented on styles/Canonical/003-Ubuntu-names-versions.yml:
```diff
@@ -0,0 +1,34 @@
+  - 'Ubuntu 19\.04 \((?!(Disco|Disco Dingo)\))'
+  - 'Ubuntu 18\.10 \((?!(Cosmic|Cosmic Cuttlefish)\))'
+  - 'Ubuntu 18\.04 LTS \((?!(Bionic|Bionic Beaver)\))'  
+
```
It would be nice to have a complete list but maybe we can make that an improvement as a straightforward ODA task

AnneCYH replied:
I can actually generate the complete list easily but it's more of a maintenance thing, e.g. it would be nice to automate this by grabbing the information from ubuntu.com.

I'll push an update in another PR.

Thanks!

evilnick reviewed (approved):
Great work - I hope my tweaks are okay with you

evilnick commented:
sorry @keirthana, could you +1 this again if you agree?

keirthana reviewed (approved):
Thanks for this, looks good
